### ISSUE TITLE
Training the reward model

### ISSUE THREAD

**Issue by farhad-abdi** [lucidrains/PaLM-rlhf-pytorch#19]
Hi, in training the reward model it seems that the 'seq' and 'propmt_mask'  should have a same length, would you be able to elaborate on training the reward model with different prompt length would it be right to do such thing:
```
        mask = torch.zeros(1,(seq[0]==prompt_id[0]).nonzero() ).bool()
        prompt_mask = torch.cat((mask[0], torch.ones(1, seq.shape[1]- mask.shape[1])[0]),0).bool().unsqueeze(0).cuda()
```

**Comment by farhad-abdi** [comment]
@lucidrains thank you! I'm  just experimenting with the model, my point was the value of padding since in other part like training palm this is zero and the mask is also zeros here.

**Comment by lucidrains** [comment]
@farhad-abdi

yeah no problem

just so we are on the same page

```bash
# P = prompt, S = generated sequence, . = padding

PPPSSS...
PPSS.....

# the corresponding prompt_mask should be

TTTFFFFFF
TTFFFFFFF

# and the mask should be

TTTTTT...
TTTT.....
```

**Comment by lucidrains** [comment]
@farhad-abdi ok, i added the `prompt_lengths` way, in case that is simpler than generating the mask yourself https://github.com/lucidrains/PaLM-rlhf-pytorch/commit/b0dfc70098a7d17151520fa3dd60ddc45600f325

**Comment by farhad-abdi** [comment]
@lucidrains  thanks for your help!
